addons/stock/models/stock_quant.py

```diff
@@ -77,7 +77,7 @@ def check_product_id(self):
     def check_quantity(self):
         for quant in self:
             if float_compare(quant.quantity, 1, precision_rounding=quant.product_uom_id.rounding) > 0 and quant.lot_id and quant.product_id.tracking == 'serial':
-                raise ValidationError(_('A serial number should only be linked to a single product.'))
+                raise ValidationError(_('The serial number %s for product %s has already been used or is used more than once in this MO. Unlock the MO to correct it.') % (self.lot_id.name, self.product_id.name))
```

Comment:
Hi, some help: 
- use `quant` and not `self` (if not you can have a singleton error), 
- use `display_name`, if case the consumer use variante of product.
- This error message is in module stock, you can not speak about MO. Add a generic message (and if possible without break existing translation).

I think the best is :
```python
raise ValidationError(_('A serial number should only be linked to a single product.') + 
'\n\n%s - %s' % (quant.product_id.display_name, quant.lot_id.name, ))
)```